### PR TITLE
[crawler] Add Module Lentille.GitHub.User

### PR DESCRIPTION
**Pull request by morucci**
Usage will be to be able to enhance Identities into Monocle with extra
info fetched from a GitHub provider.

### CODE REVIEW

File: haskell/src/Lentille/GitHub/User.hs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import Data.Morpheus.Client
+import Lentille
+import Lentille.GitHub.RateLimit (retryCheck)
+import Lentille.GraphQL
+import Monocle.Prelude
+
+newtype DateTime = DateTime Text deriving (Show, Eq, EncodeScalar, DecodeScalar)
+
+data IdentInfo = IdentInfo
+  { iiName :: Maybe Text,
```
Comment (TristanCacqueray) [inline haskell/src/Lentille/GitHub/User.hs:17]: Shouldn't this be:

```suggestion
  { iiName :: Text,
```

File: haskell/src/Lentille/GitHub/User.hs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        resetAt
+      }
+      user(login: $login) {
+        name
+        company
+      }
+    }
+  |]
+
+transformResponse :: GetUser -> (Maybe RateLimit, IdentInfo)
```
Comment (TristanCacqueray) [inline haskell/src/Lentille/GitHub/User.hs:37]:
```suggestion
transformResponse :: GetUser -> (RateLimit, Maybe IdentInfo)
```

File: haskell/src/Lentille/GitHub/User.hs (hunk 1 above)
Comment (morucci) [reply haskell/src/Lentille/GitHub/User.hs]: The schema binds to a Maybe Text

File: haskell/src/Lentille/GitHub/User.hs (hunk 2 above)
Comment (morucci) [reply haskell/src/Lentille/GitHub/User.hs]: Yes, I've removed the Maybe RateLimit. However we should get a IdentInfo or an exception will be raised, so for me that's an IdentInfo but not a Maybe IdentInfo.

**Comment by morucci** [comment]
@TristanCacqueray could I get your approval ?

**Review by TristanCacqueray — APPROVED** [review]

**Pull request merged**